client/src/main/java/com/tigrisdata/db/client/TypeConverter.java

```diff
@@ -145,14 +146,39 @@ public static Api.ReadRequest toReadRequest(
         Api.ReadRequest.newBuilder()
             .setDb(databaseName)
             .setCollection(collectionName)
-            .setFilter(ByteString.copyFrom(filter.toJSON(objectMapper), StandardCharsets.UTF_8))
+            .setFilter(ByteString.copyFromUtf8(filter.toJSON(objectMapper)))
             .setOptions(readRequestOptionsAPI);
     if (!fields.isEmpty()) {
       readRequestBuilder.setFields(ByteString.copyFromUtf8(fields.toJSON(objectMapper)));
     }
     return readRequestBuilder.build();
   }
 
+  public static Api.SearchRequest toSearchRequest(
+      String databaseName, String collectionName, SearchRequest req, ObjectMapper objectMapper) {
+    Api.SearchRequest.Builder builder =
+        Api.SearchRequest.newBuilder()
+            .setDb(databaseName)
+            .setCollection(collectionName)
+            .setQ(req.getQuery().toJSON(objectMapper));
+    if (Objects.nonNull(req.getSearchFields())) {
+      builder.addAllSearchFields(req.getSearchFields().get());
+    }
+    if (Objects.nonNull(req.getFilter())) {
+      builder.setFilter(ByteString.copyFromUtf8(req.getFilter().toJSON(objectMapper)));
+    }
+    if (Objects.nonNull(req.getFacetQuery())) {
+      builder.setFacet(ByteString.copyFromUtf8(req.getFacetQuery().toJSON(objectMapper)));
+    }
+    if (Objects.nonNull(req.getSortOrders())) {
+      builder.setSort(ByteString.copyFromUtf8(req.getSortOrders().toJSON(objectMapper)));
+    }
+    if (Objects.nonNull(req.getReadFields())) {
```

Comment:
`&& !req.getReadFields().isEmpty()` ?

Reply:
will be refactoring the Read fields altogether